packages/@uppy/image-editor/src/ImageEditor.tsx

```diff
@@ -89,12 +89,21 @@ const defaultOptions = {
   quality: 0.8,
   actions: defaultActions,
   cropperOptions: defaultCropperOptions,
-} satisfies Opts
-
-export type ImageEditorOpts = DefinePluginOpts<
-  Opts,
-  keyof typeof defaultOptions
->
+} satisfies Partial<Opts>
+
+export type ImageEditorOpts = Omit<
+  DefinePluginOpts<Opts, keyof typeof defaultOptions>,
+  'actions' | 'cropperOptions'
+> & {
+  actions: DefinePluginOpts<
+    NonNullable<Opts['actions']>,
+    keyof typeof defaultActions
+  >
+  cropperOptions: DefinePluginOpts<
+    NonNullable<Opts['cropperOptions']>,
+    keyof typeof defaultCropperOptions
+  >
+}
```

Comment:
Can't we in this case remove the `Partial`'s above so this type can stay simple? This seems a bit intense.

Reply:
The issue is that we pass `keyof typeof default…`, so there's no way to validate the values of the object, only the keys. If there was a way to pass `typeof default…` instead, we would no longer need the `satisfies` above. I tried the following which didn't work (`Type 'keyof DefaultValues' does not satisfy the constraint 'keyof Opts'.`):

```typescript
export type DefinePluginOpts<
  Opts,
  DefaultValues extends OnlyOptionals<Opts>,
> = Opts & Required<Pick<Opts, keyof DefaultValues>>
```

Reply:
> `ImageEditorOpts` should […] not be exported

Good catch, I've renamed it `InternalImageEditorOpts`.



> So you're saying this is only a problem because we have nested object types here with `actions` and such? The problem of "no way to validate the values of the object" is not present elsewhere?

That's not what I'm saying, although having nested objects certainly makes the types more complicated.
Currently we need the `satisfies` to ensure the default values match the types of the associated option. Maybe there's a way to validate that somewhere else, in a more transparent way, but I wasn't able to find a way to do that.

Reply:
I checked it out locally and and removed the `Partial`'s and got no errors. Returning to the simple type of `InternalImageEditorOpts` also gave no errors. Where do we see the effect of this more complex type then?

Reply:
Well of course you won't see any error if you have looser types 🤷‍♂️ the idea is to have TS report when we use an incompatible type (i.e. you should test by changing the value to something incompatible, e.g. 
<img width="299" alt="image" src="https://github.com/transloadit/uppy/assets/14309773/821423f7-b572-435e-8d35-fc463b123af0">)